nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -564,14 +736,370 @@ impl DataStore {
+                                            as i64
+                                            + request
+                                                .request
+                                                .required_dataset_overhead()
+                                                .to_bytes()
+                                                as i64;
+                                    request_size
+                                },
+
+                                local_storage_dataset_id: candidate_dataset
+                                    .rendezvous_local_storage_dataset_id,
+
+                                pool_id: candidate_dataset.pool_id,
+
+                                sled_id: candidate_dataset.sled_id,
+                            });
+
+                            // Note by removing a candidate dataset from the
+                            // list in this way, this step mandates that a
+                            // single local storage dataset is not used for
+                            // multiple local storage allocations for a given
+                            // instance.
+
+                            let mut set_candidates_left =
+                                candidates_left.clone();
+                            set_candidates_left.remove(candidate_dataset);
+
+                            queue.push_back(PossibleAllocations {
+                                allocations: set_allocations,
+                                candidates_left: set_candidates_left,
+                                request_index: request_index + 1,
+                            });
+                        }
+
+                        // Else there are no candidate datasets left for this
+                        // request, and therefore the list of requests cannot be
+                        // fulfilled by the current mapping of requests to
+                        // datasets
+                    }
+                }
+
+                // Loop here over each possible set of local storage allocations
+                // required, and attempt the sled insert resource query with
+                // that particular allocation.
+                //
+                // If the `validated_allocations` list is empty, control will
+                // pass to the end of the loop marked with 'outer, which will
+                // then try the next possible sled target. In the case where
+                // there were existing local storage allocations there will
+                // _not_ be any more sleds to try and the user will see a
+                // capacity error.
+
+                for valid_allocation in validated_allocations {
+                    let ValidatedAllocations { allocations } = valid_allocation;
+
+                    info!(
+                        &log,
+                        "attempting to insert sled resource record";
+                        "local storage required" => "yes",
```

Comment:
```suggestion
                        "local_storage_required" => true,
```
if we make the change i suggested above